tree/ntuplebrowse/v7/inc/ROOT/RBrowseVisitor.hxx

```diff
@@ -0,0 +1,134 @@
+   void VisitRootField(const RFieldRoot &/*field*/, int /*level*/) final { }
+};
+
+   
+   
+// clang-format off
+/**
+\class ROOT::Experimental::RDisplayHistoVisitor
+\ingroup NTupleBrowse
+\brief Visitor class which draws a histogram for fields with numerical data.
+    
+ Visits fields displayed in TBrowser and draws a histogram for appropriate fields. Instances of this class are created when a field without subfields is double-clicked in TBrowser. (called by RNTupleBrowseLeaf::Browse(TBrowser* b))
+*/
+// clang-format on
+class RDisplayHistoVisitor: public Detail::RNTupleVisitor {
+private:
+   /// Allows to access RNTupleBrowser::fCurrentTH1F.
+   RNTupleBrowser*      fNTupleBrowserPtr;
+   /// Allows to get entries of a field which will be displayed in the histogram.
+   RNTupleReader*       fNTupleReaderPtr; // Note: fNTupleBrowserPtr->GetReaderPtr() returns the last created RNTupleReader. This can be another RNTupleReader than the one which contains information about the visited field. Therefore a separate member had to be created.
+
+public:
+   RDisplayHistoVisitor(RNTupleBrowser* parntplb, RNTupleReader* readerPtr):
+      fNTupleBrowserPtr{parntplb},
+      fNTupleReaderPtr{readerPtr}
+      {}
+   
+   void VisitField(const Detail::RFieldBase &/*field*/, int /*level*/) final { }
+   void VisitRootField(const RFieldRoot &/*field*/, int /*level*/) final { }
+   void VisitFloatField(const RField<float> &field, int /*level*/) { DrawHistogram<float>(field, false); }
+   void VisitDoubleField(const RField<double> &field, int /*level*/) { DrawHistogram<double>(field, false); }
+   void VisitInt32Field(const RField<std::int32_t> &field, int /*level*/) { DrawHistogram<std::int32_t>(field, true); }
+   void VisitUInt32Field(const RField<std::uint32_t> &field, int /*level*/) { DrawHistogram<std::uint32_t>(field, true); }
+   void VisitUInt64Field(const RField<std::uint64_t> &field, int /*level*/) { DrawHistogram<std::uint64_t>(field, true); }
+   
+   template <typename T>
+   void DrawHistogram(const Detail::RFieldBase &field, bool isIntegraltype)
+   {
+      // only leaf-fields should display a histogram.
+      if (field.GetStructure() != kLeaf)
+         return;
+      // Currently subfield of a vector field shouldn't display a histogram. TODO (lesimon): think if it should be displayed and if so how.
+      if (std::string(field.GetParent()->GetType(), 0, 12).compare("std::vector<") == 0)
+         return;
+      
+      // for now only print fields directly attached to RootField, until RNTupleView is fixed. TODO(lesimon): Remove this later.
+      if (field.GetLevelInfo().GetLevel() != 1)
+         return;
+      
+      auto ntupleView = fNTupleReaderPtr->GetView<T>(field.GetName());
+      
+      // if min = 3 and max = 10, a histogram with a x-axis range of 3 to 10 is created with 8 bins (3, 4, 5, 6, 7, 8, 9, 10)
+      double max{LONG_MIN}, min{LONG_MAX};
+      for (auto i : fNTupleReaderPtr->GetViewRange()) {
```

Comment:
I thought about it for a time and the for loop is necessary for filling the histogram. So there needs to be a way to determine the x-axis limits (e.g. min/max) of the histogram in a different way. One way would be with random sampling, but it would be strange if the histogram looked slightly different every time you generate it.